extensions/machine-learning-services/src/packageManagement/packageManager.ts

```diff
@@ -0,0 +1,166 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as nbExtensionApis from '../typings/notebookServices';
+import { SqlPythonPackageManageProvider } from './sqlPackageManageProvider';
+import { QueryRunner } from '../common/queryRunner';
+import * as utils from '../common/utils';
+import * as constants from '../common/constants';
+import { ApiWrapper } from '../common/apiWrapper';
+import { Deferred } from '../common/promise';
+
+// Required python packages for mls extension
+//
+const requiredPackages = [
+	{ name: 'pymssql', version: '2.1.4' },
+	{ name: 'sqlmlutils' }
+];
```

Comment:
just for my understanding... is defining package here the common way? there is not config way?

Reply:
I'd suggest looking at what @alanrenmsft did for the deployment dialog - that's done in the config and is definitely a better approach (and makes it so other extensions can also contribute if we desire that functionality)

Reply:
what do you mean by config way? I'll see what @chgagnon@microsoft.com  is talking about and maybe I can improve this

---
In reply to: [356365306](https://github.com/microsoft/azuredatastudio/pull/8622#discussion_r356365306) [](ancestors = 356365306)

Reply:
I talked to him but the thing I'm doing here is different than what he did. I can explain in person

---
In reply to: [356683991](https://github.com/microsoft/azuredatastudio/pull/8622#discussion_r356683991) [](ancestors = 356683991)

Reply:
That's fine - but is there a reason you wouldn't have this be something in a config regardless? Even if it isn't a contributed config having a separate config that's loaded at runtime seems like a much better way to store stuff like this. For example this is what we do for the server downloader : https://github.com/microsoft/azuredatastudio/blob/master/extensions/mssql/config.json

Reply:
sure I can move this to a config file

---
In reply to: [357287750](https://github.com/microsoft/azuredatastudio/pull/8622#discussion_r357287750) [](ancestors = 357287750)